cl/lib/elasticsearch_utils.py

```diff
@@ -953,17 +974,15 @@ def merge_highlights_into_result(
                 unique_marked_strings = list(
                     set(marked_strings + marked_strings_exact)
                 )
-                combined_highlights = highlight_list[0]
-                for marked_string in unique_marked_strings:
-                    # Replace unique highlighted terms in a single
-                    # field.
-                    unmarked_string = marked_string.replace(
-                        f"<{tag}>", ""
-                    ).replace(f"</{tag}>", "")
-                    combined_highlights = combined_highlights.replace(
-                        unmarked_string, marked_string
-                    )
-
+                original_string = highlight_list[0]
+                all_terms_pattern = rf"<{tag}>|</{tag}>|\s+|[^\s<]+"
```

Comment:
Do we really need to match whitespaces in the original string?

Reply:
Yes, I rechecked the regex pattern and you're right, matching whitespaces is not required. We only need to match the HL tags and words. So, I've updated the pattern to  `"<{tag}>|</{tag}>|\w+"` as suggested.